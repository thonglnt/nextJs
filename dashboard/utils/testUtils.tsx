import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
