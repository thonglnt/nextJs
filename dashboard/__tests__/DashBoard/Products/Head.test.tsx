// Utils
import { render } from '@utils/testUtils'

// App
import Head from 'app/(dashboard)/products/head'

describe('Head dash board render', () => {
  it('Should Head match Snapshot', () => {
    const { container } = render(<Head />)

    expect(container).toMatchSnapshot()
  })
})
