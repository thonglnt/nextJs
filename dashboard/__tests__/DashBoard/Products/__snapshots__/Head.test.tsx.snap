// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head dash board render Should Head match Snapshot 1`] = `
<div>
  <title>
    Products | Material
  </title>
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <link
    href="/favicon.ico"
    rel="icon"
  />
</div>
`;
